cmd/gateway-main.go

```diff
@@ -95,6 +95,8 @@ func mustGetGatewayCredsFromEnv() (accessKey, secretKey string) {
 	if accessKey == "" || secretKey == "" {
 		fatalIf(errors.New("Missing credentials"), "Access and secret keys are mandatory to run Minio gateway server.")
 	}
+	globalIsEnvCreds = true
```

Comment:
Add a comment.. here @poornas - perhaps in the unit test you can check for this value to be set appropriately?